### PR TITLE
fix(sql-execute): no trace id when sql executing failed

### LINKED ISSUE #697: [Bug]: Executing sql does not show time consuming
### ODC version

ODC 4.2.2

### OB version

OB 4.2.1

### What happened?

Executing the following sql does not show time consuming.
```SQL
SELECT
      TABLE_OWNER,
      TABLE_NAME,
      COLUMN_POSITION,
      COLUMN_NAME,
      INDEX_NAME
    FROM
      ALL_IND_COLUMNS
    WHERE
      TABLE_OWNER = 'GSH' AND TABLE_NAME = 'TEST_INDEX1'
    ORDER BY
      COLUMN_POSITION ASC
```
image

### What did you expect to happen?

Display the time consuming.

### How can we reproduce it (as minimally and precisely as possible)?

Follow What happened.

### Anything else we need to know?

_No response_

### Cloud

_No response_

### PR DESCRIPTION
**Pull request by LuckyPickleZZ**
#### What type of PR is this?
type-bug

#### What this PR does / why we need it:
When sql executing failed, ODC would not query trace id for it. In this PR, there will always be trace id, no matter it's succeeded or failed.

#### Which issue(s) this PR fixes:

Fixes #697

### CODE REVIEW

File: server/odc-core/src/main/java/com/oceanbase/odc/core/sql/util/FullLinkTraceUtil.java [hunk 1]
```diff
@@ -48,7 +48,12 @@ public static SqlExecTime getFullLinkTraceDetail(Statement statement) throws SQL
                 throw new UnexpectedException("No trace info, maybe value of ob_enable_show_trace is 0.");
             }
             String showTraceJson = resultSet.getString(1);
-            SqlExecTime execDetail = parseSpanList(JsonUtils.fromJsonList(showTraceJson, TraceSpan.class));
+            SqlExecTime execDetail;
+            try {
+                execDetail = parseSpanList(JsonUtils.fromJsonList(showTraceJson, TraceSpan.class));
+            } catch (Exception e) {
+                throw new UnexpectedException("Parse trace failed, original text: " + showTraceJson, e);
```
Comment (yhilmare) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/sql/util/FullLinkTraceUtil.java:55]: showTraceJson will be too large, don't print it directly

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java [hunk 2]
```diff
@@ -176,6 +176,9 @@ public List<JdbcGeneralResult> doInStatement(Statement statement) throws SQLExce
                 if (thrown.get() == null || !stopWhenError) {
                     try {
                         executeResults = doExecuteSql(statement, sqlTuple);
+                        if (executeResults.size() == 1 && executeResults.get(0).getThrown() != null) {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java:179]: there is a filed called `status` in `JdbcGeneralResult` that can be used to judge if the execution is succeed

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java [hunk 3]
```diff
@@ -331,7 +341,14 @@ protected List<JdbcGeneralResult> doExecuteSql(Statement statement, SqlTuple sql
                     throw new NotImplementedException("Unsupport function call " + definition.getFunctionName());
                 }
             }
-            boolean isResult = preparedStatement.execute();
+            boolean isResult;
+            try {
+                isResult = preparedStatement.execute();
+            } catch (Exception e) {
+                JdbcGeneralResult failedResult = JdbcGeneralResult.failedResult(sqlTuple, e);
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java:348]: duplicate code

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java [hunk 4]
```diff
@@ -390,7 +407,7 @@ private String getTraceIdAndAndSetStage(Statement statement, TraceWatch traceWat
             setExecuteTraceStage(traceWatch, executeDetails, stopWatch);
             return executeDetails.getTraceId();
         } catch (Exception ex) {
-            log.warn("Query sql execute details failed, reason={}", ex.getMessage());
+            log.warn("Query sql execute details failed.", ex);
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java:410]: there is no need to print all err stacks

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java (hunk 2 above)
Comment (LuckyPickleZZ) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java:179]: this field is not usable. The `status` will only be failed when the code enters line 192.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java (hunk 3 above)
Comment (LuckyPickleZZ) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java]: done, removed .

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java (hunk 4 above)
Comment (LuckyPickleZZ) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java]: done

File: server/odc-core/src/main/java/com/oceanbase/odc/core/sql/util/FullLinkTraceUtil.java (hunk 1 above)
Comment (LuckyPickleZZ) [reply server/odc-core/src/main/java/com/oceanbase/odc/core/sql/util/FullLinkTraceUtil.java]: done.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java (hunk 4 above)
Comment (LuckyPickleZZ) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java]: done

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java (hunk 2 above)
Comment (LuckyPickleZZ) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java]: done , fixed this.

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**